provider/aws/sia-ec2/authn.go

```diff
@@ -128,7 +128,9 @@ func GetEC2AccessProfile(configFile, metaEndpoint string, useRegionalSTS bool, r
 			// to use fallback to <domain>.<service>-service@access-profile naming structure
 			log.Println("Trying to determine profile name from security credentials...")
 			_, accessProfileConfig, err = options.InitCredsConfig("-service", "@", useRegionalSTS, region)
-			if err != nil {
+			// if either the profile is empty which can happen when assumed role doesn't have access profile info
+			// or if error is not nil try to determine access profile info from instance profile
+			if err != nil || accessProfileConfig == nil || accessProfileConfig.Profile == "" {
```

Comment:
the error line 134 in this block seems to be wrong as well. First, we're not processing the configuration file here - it's the credentials. Next since err can be nil, we should just include a generic error that we couldn't obtain the profile details from the security credentials.